### PR TITLE
Fix bug: do not skip instructions that return types that aren't taintable

### PR DESCRIPTION
**Pull request by mlevesquedion**
This fixes a bug discovered in #287. See that PR for details.

TLDR: When checking whether we should avoid a given `SSA` node based on its type (e.g., `bool` values can't be tainted), we should still do propagate to the operands if we normally would. We should avoid the node's referrers though, because that can lead to incorrect propagations. See the test case added by this PR for an example.

- [x] Running against a large codebase such as [Kubernetes](https://github.com/kubernetes/kubernetes) does not error out. (See [DEVELOPING.md](https://github.com/google/go-flow-levee/blob/master/DEVELOPING.md) for instructions on how to do that.)
- (N/A) [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Review by PurelyApplied — APPROVED** [review]
Thanks for extracting this.
